src/baldr/tilehierarchy.cc

```diff
@@ -6,21 +6,25 @@ using namespace valhalla::midgard;
 namespace valhalla {
 namespace baldr {
 
-const std::map<uint8_t, TileLevel>& TileHierarchy::levels() {
+const std::vector<TileLevel>& TileHierarchy::levels() {
```

Comment:
so i just couldnt let this stupidity linger. while chasing down a failing test, i decided to refactor the hierarchy static structure. there is no reason this should be a map. instead its a vector where the index is the level. why do binary search on something that you already know where the indices should be? this caused a lot of diffs in the rest of this PR. if you see anything with `TileHeirarchy::` it was this refactor